nix/ibis.nix

```diff
@@ -1,61 +1,46 @@
-{ poetry2nix
-, python3
-, lib
-, gitignoreSource
-, graphviz-nox
-, sqlite
-, ibisTestingData
-}:
-# pyspark could be added here, but it doesn't handle parallel test execution
-# well and serially it takes on the order of 7-8 minutes to execute serially
+{ pkgs, python, mkEnv, stdenv, ... }:
 let
-  extras = [ "decompiler" "visualization" ];
-  backends = [ "duckdb" "polars" "sqlite" ];
-in
-poetry2nix.mkPoetryApplication {
-  python = python3;
-  groups = [ "main" ];
-  checkGroups = [ "main" "test" ];
-  projectDir = gitignoreSource ../.;
-  src = gitignoreSource ../.;
-  extras = backends ++ [ "datafusion" ] ++ extras;
-  overrides = [
-    (import ../poetry-overrides.nix)
-    poetry2nix.defaultPoetryOverrides
-  ];
-  preferWheels = true;
-  __darwinAllowLocalNetworking = true;
-
-  POETRY_DYNAMIC_VERSIONING_BYPASS = "1";
-
-  nativeCheckInputs = lib.optionals (lib.elem "sqlite" backends) [ sqlite ]
-    ++ lib.optionals (lib.elem "visualization" extras) [ graphviz-nox ];
+  pythonEnv = mkEnv python {
+    editable = false;
+    deps = {
+      ibis-framework = [
+        # Groups
+        "tests"
+        # Extras
+        "duckdb"
+        "datafusion"
+        "sqlite"
+        "polars"
+        "decompiler"
+        "visualization"
+      ];
+    };
+  };
 
+in
+stdenv.mkDerivation {
+  name = "ibis-framework-test";
+  nativeCheckInputs = [ pythonEnv pkgs.graphviz-nox ];
+  src = ../.;
+  doCheck = true;
   preCheck = ''
-    set -euo pipefail
-
-    HOME="$(mktemp -d)"
-    export HOME
-
-    ln -s "${ibisTestingData}" $PWD/ci/ibis-testing-data
+    ln -s ${pkgs.ibisTestingData} $PWD/ci/ibis-testing-data
+  '';
+  checkPhase = ''
+    runHook preCheck
+    pytest -m datafusion
+    pytest -m 'core or duckdb or sqlite or polars' --numprocesses $NIX_BUILD_CORES --dist loadgroup
+    runHook postCheck
   '';
 
-  checkPhase =
-    let
-      markers = lib.concatStringsSep " or " (backends ++ [ "core" ]);
-    in
-    ''
-      set -euo pipefail
-
-      runHook preCheck
-
-      pytest -m datafusion
-      pytest -m '${markers}' --numprocesses $NIX_BUILD_CORES --dist loadgroup
-
-      runHook postCheck
-    '';
-
-  doCheck = true;
-
-  pythonImportsCheck = [ "ibis" ] ++ map (backend: "ibis.backends.${backend}") (backends ++ [ "datafusion" ]);
+  # Ibis-framework was already built as a part of the env, this is just running
+  # tests. Symlink the built test env for convenience.
+  #
+  # Note: Testing could technically be done as a part of the virtualenv
+  # constructor derivation.
+  installPhase = ''
+    runHook preInstall
+    cp -r ${pythonEnv} $out
```

Comment:
Symlinking this should work with `dontFixup = true`.

Reply:
Ah, right, thank you!

Reply:
Fixed!